File: robin/java/src/level1/study3/PocketMon.java
package level1.study3;

import java.util.HashSet;
import java.util.Set;

public class PocketMon {

    public static int solution(int[] nums) {

        int maxPocketMon = nums.length / 2;

        Set<Integer> hs = new HashSet<>();

        for (int num : nums) {
            hs.add(num);
        }

        return Math.min(maxPocketMon, hs.size());
    }

    public static void main(String[] args) {
        int[] nums = {3, 1, 2, 3};
        System.out.println(solution(nums));
    }
}

File: robin/python/level1/study3/pocketmon.py
def solution(nums):
    answer = 0
    pocket_num = len(nums) / 2

    set_nums = set(nums)

    if pocket_num < len(set_nums) or pocket_num == len(set_nums):
        answer = pocket_num

    if pocket_num > len(set_nums):
        answer = len(set_nums)

    return answer


def solution2(nums):
    return min(len(nums) / 2, len(set(nums)))


if __name__ == '__main__':
    nums = [3, 1, 2, 3]  # 2
    nums = [3, 3, 3, 2, 2, 4]  # 3
    # nums = [3, 3, 3, 2, 2, 2] #2
    print(solution(nums))
    print(solution2(nums))
